requests/utils.py

```diff
@@ -665,6 +665,21 @@ def parse_header_links(value):
 
     return links
 
+def is_valid_location(response):
+    """Verify that multiple Location headers weren't
+    returned from the last response.
+    """
+    if hasattr(response.raw, 'headers'):
+        if hasattr(response.raw.headers, 'getlist'):
+            return len(response.raw.headers.getlist('location')) <= 1
+        else:
+            pass # Do something with header value
+    elif hasattr(response.raw, 'read'):
+        response.raw.seek(0) # Move to beginning of file
+        data = response.raw.read()
+        # parse_headers(data) # try to parse headers out of raw response
```

Comment:
I think this is more than we need. If we can't get headers in the above case, don't worry about it. 

Mostly what I'm concerned about here is not exploding in the non-urllib3 case: we don't need to be too clever, or even really "handle" it in any meaningful way.